### ISSUE TITLE
Feature Request: Include loss_ as an attribute for the fitting

### ISSUE THREAD

**Issue by jolespin** [lmcinnes/umap#174]
First off, thank you for implementing this method in Python!  Very stoked to start using it for my bioinformatics datasets.  I have been trying to quantify which parameters are the best for my datasets and having some trouble.  I was wondering if the loss provided in your Enthought talk from SciPy 2018: 
![image](https://user-images.githubusercontent.com/9061708/49530699-97ff1300-f86d-11e8-9d66-f4a596b4d786.png)

Could be included as an attribute we can access later so I can figure out which of my hyperparameters settings should be used? 

I have a precomputed data matrix `(137 x 137)` and using the following hyperparameter configs:
```python
for n_neighbors in [3,4,5,6,7]:
    for min_dist in [0.01, 0.1,0.2,0.3,0.5]:
        for spread in [0.01, 0.1,0.2,0.3,0.5]:
            for learning_rate in [1e-3, 1e-2,1e-1,1]:
```

There is some structure that fits my hypothesis from some of these configs and I want to know which one specifically I should choose so I thought maybe so sort of `loss_` metric would be really useful in this scenario.

Also. if you have a few moments.  Can you describe why I see this topology sometimes where it looks like a regression?
![image](https://user-images.githubusercontent.com/9061708/49532094-9b47ce00-f870-11e8-9de7-4d5554e5dfe8.png)
![image](https://user-images.githubusercontent.com/9061708/49532144-b9153300-f870-11e8-8545-715fafbcd96c.png)

**Comment by lmcinnes** [comment]
Computing the full loss is potentially computationally nightmarish for large datasets, which is why it is not currently done. As for your current issue with weird structures appearing -- I believe that is due to a bug that was very recently fixed (see issue #170 ). If you upgrade to the latest version on pip, or install directly from the current master branch on github it should fix that problem.

**Comment by jolespin** [comment]
I installed via `pip install git+https://github.com/lmcinnes/umap` a few days ago.  I found some other parameter combinations that have very interpretable results so I will stick with those :) 
![image](https://user-images.githubusercontent.com/9061708/49608694-29de4d00-f94e-11e8-95db-727bbd58d3d5.png)

Do you have any suggestions on ways to compare hyperparamter methods? I've manually gone through about 500 parameter configs and definitely noticed a pattern.  The only thing I could think of was to cluster and do a silhouette score but is there anything I could grab from the actual model that could quantify that params_A are better than params_B?

**Comment by jolespin** [comment]
Thanks, this is really helpful.  What range of learning_rates would you recommend if one were to adjust these?

**Comment by jolespin** [comment]
Thanks, this is really helpful. What range of `learning_rates` would you recommend if one were to adjust these?

**Comment by lmcinnes** [comment]
I would start with a learning rate of 1.0 (the default). You can scale it down a bit if you need, but realistically you shouldn't need to get too much smaller than 0.5 I would imagine.
